### ISSUE TITLE
cannot understand how the config is included

### ISSUE THREAD

**Issue by duck-rh** [OSAS/ansible-role-httpd#14]
I just switched to tls for my site and now both vhost.conf files, TLS and non-TLS are including letsencrypt.conf but no more vhost.conf.d/*conf. letsencrypt.conf does not include then either. I've got another unrelated problem but it seems my redirects are not working because of this. I checked the default inclusion directives but there is not way it would include these files (and no way it could be magically into the right vhost anyway).

I guess letsencrypt.conf is included here for ordering reasons.

I'm not really sure what was the intended logic, so I don't really know yet what PR to do.

I can do the actual fix if information is provided.

**Comment by mscherer** [comment]
So, I can't exactly understand what is the issue, cause you give neither the symptoms, nor the ansible configuration you used, or the version of the role. In fact, I am not sure also what information to give as there is no question.

I would be ok to provides answers if there is questions.
So:
- what is the playbook used (or just the httpd part)
- what is the outcome you want
- what is the outcome you see

Or:
- what is the question, or part that can't be understood regarding include.

**Comment by duck-rh** [comment]
I'm talking about the Apache inclusion directives in the role generated files, and I don't see any which would allow including the files in the <vhost>.conf.d directory.
I'm using the role's 'redirects' feature and the one I added is not working.
So my question is how vhost.conf inclusions are supposed to work because it does not seem to be defined anywhere. Also the manual inclusion of letsencrypt.conf seems to be on purpose,so I'd like to know why.
So it seems I finally understood at least part of it, and there is a bug, that's why I proposed a PR.

It is used for the Cage like this:
```
  - role: httpd
    website_domain: "{{ ansible_fqdn }}"
    redirects:
    - src: '^/$'
      target: '/rt'
      match: True
    use_letsencrypt: True
    force_tls: True
```

**Comment by mscherer** [comment]
So here is the trick. We found earlier that inclusion of everything for every vhost didn't made any sense.

Some settings are taking over the complete vhost (or rather taking over vhost, or vhost + port).
For example, redirect is redirecting a complete vhost, force_tls does it too. So I moved force_tls in vhost.conf directly, as a attempt to provides sanity. cf https://github.com/OSAS/ansible-role-httpd/pull/6
(didn't yet for the other, by lack of time and because I wanted to make sure this did work)

Now, maybe you did hit a bug indeed and reading the code before coffee, I can indeed see clearly a issue (ie, that it do not verify that the port is 443 before doing the redirection). 

And yes, the commit bc4315d735c49c28eef of #15 is the fix I would use (even if I think the hardcoding of port we have in vhost is bad, but we can just decide that this manage just port 80 and 443 for now).

(I would also put a better commit message, like at least trying to explain what was the problem)

But I still do not understand the 2nd party of the patch, but that's something we can discuss on #15.

**Comment by duck-rh** [comment]
I think your fix attempt is fine, as there is nothing I see we would need of the included config inside the redirected vhost, so only letsencrypt is fine to me. The only thing I would have done differently is using a different way of defining which include is to be always there using symlinks in another directory, so as to allow external tasks or roles to have their own reason to do it is necessary. But I don't see any real life need now, so it's simple and fine.

The second commit is unrelated, as stated in the description: « Please note PR #15 has to be resolved first; I needed this stack of fixes to have my current work in shape. ». This is my way to go on working, and if you do not agree then I'll have to rebase all the pile along, which is the risk I agreed with myself to take.

As for using the ports, I figured out reading the rest of the code there is no support for other ports, so I just duplicated the way of doing; if we want to improve, I'm totally fine with this, then that's another PR.

**Comment by mscherer** [comment]
Mhh nice, the simple fact to say "fix #14" close the bug 

And yeah, I am not sure if we need to support others ports. That's cleaner in theory, but if we do not use, that's not needed. I just list that so I can say "I knew it" when later we will have the need. I am not in favor of making thing too complciated for a case we do not have yet.
